src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -1955,6 +1955,61 @@ public function get_test_rest_availability() {
 		return $result;
 	}
 
+	/**
+	 * Test if 'file_uploads' directive in PHP.ini is turned off
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array The test results.
+	 */
+	public function get_test_file_uploads() {
+		$result = array(
+			'label'       => __( 'File uploads is on in PHP ini' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Media' ),
+				'color' => 'blue',
+			),
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( 'The <code>file_uploads</code> directive in <code>php.ini</code> determines if uploading to is allowed in your WordPress.' )
+			),
+			'actions'     => '',
+			'test'        => 'file_uploads',
+		);
+
+		if ( ! function_exists( 'ini_get' ) ) {
+			$result['status']       = 'critical';
+			$result['description'] .= sprintf(
+				/* translators: %s: ini_get() */
+				__( 'Unable to determine some settings, as the %s function has been disabled.' ),
+				'ini_get()'
+			);
+			return $result;
+		}
+
+		if ( empty( ini_get( 'file_uploads' ) ) ) {
+			$result['status']       = 'critical';
+			$result['description'] .= sprintf(
+				'<p>%s</p>',
+				__( '<code>file_uploads</code> is set to <code>0</code>. You won\'t be able to upload files in your WordPress.' )
+			);
+			return $result;
+		}
+
+		if ( $this->parse_ini_size( ini_get( 'post_max_size' ) ) !== $this->parse_ini_size( ini_get( 'upload_max_filesize' ) ) ) {
+			$result['label']       = __( 'Mismatched "post_max_size" and "upload_max_filesize" values.' );
+			$result['status']      = 'recommended';
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Mismatched values for PHP INI directives <code>post_max_size</code> and <code>upload_max_filesize</code> may cause problems. These values are in PHP INI.' )
```

Comment:
This also needs to separate out markup.

Can also be simplified a little I think, as the site health checks are more "public facing", the language needs to be made as non-technical as possible.

`The settings for <code>post_max_size</code> and <code>upload_max_filesize</code> are not the same, this could cause some problems when trying to upload files.

It might make sense to add an action linking to an official document on how to change these values at WordPress.org (if none exist, reach out to the #docs team, and they can help get one created for this).